### PR TITLE
Feature: Save archive creation settings

### LINKED ISSUE #16340: Feature: Save archive creation settings
### What feature or improvement do you think would benefit Files?

Currently they reset to defaults and have to be changed again. It would be nice to save these and could also be used for the preset options

### Requirements

- Save settings for 7z and zip creation in the `create archive` menu
- Optionally apply these settings when clicking the presets.

### Files Version

3.7.11.0

### Windows Version

10.0.22631.4169

### Comments

_No response_

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #16340

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Confirmed the selected archive creation options are correctly saved

### CODE REVIEW

File: src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs [hunk 1]
```diff
@@ -107,6 +107,7 @@ private void ViewModel_PropertyChanged(object? _, PropertyChangedEventArgs e)
 
 		private sealed class DialogViewModel : ObservableObject
 		{
+			private readonly IGeneralSettingsService GeneralSettingsService = Ioc.Default.GetRequiredService<IGeneralSettingsService>();
```
Comment (0x5bfa) [inline src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs:110]:
```suggestion
			private readonly IGeneralSettingsService GeneralSettingsService = Ioc.Default.GetRequiredService<IGeneralSettingsService>();

```

File: src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs [hunk 2]
```diff
@@ -125,33 +126,41 @@ public string FileName
 				}
 			}
 
-			private FileFormatItem fileFormat;
 			public FileFormatItem FileFormat
 			{
-				get => fileFormat;
+				get => FileFormats.First(format => format.Key == GeneralSettingsService.ArchiveFormatsOption);
```
Comment (0x5bfa) [inline src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs:131]: Is this null-safe?

Comment (yaira2) [reply src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs:131]: It should be

Comment (0x5bfa) [reply src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs:132]: I mean does C# code analysis show warning for it? We'd like to reduce numbers of possible null warnings. Sorry I can't access my PC

Comment (yaira2) [reply src/Files.App/Dialogs/CreateArchiveDialog.xaml.cs:132]: No

**Pull request merged**